### PR TITLE
feat: orchard parent tree api

### LINKED ISSUE #119: Orchard Parent Tree api
Given an Orchard ID and a SPU ID, retrieve the parent tree and genetic quality information. There's a Jira ticket that you can see it here: https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-533

### PR DESCRIPTION
**Pull request by RMCampos**
# Description

Create resource for retrieving Orchard's Parent Tree and Parent Tree Genetic Quality data.

Fixes #119 and [This Jira Ticket](https://apps.nrs.gov.bc.ca/int/jira/browse/FSADT2-533)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This was tested locally, with unit testing, but also running it locally and consuming this API.

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have already been accepted and merged

## Further comments

You can test this feature with `Orchard ID=405` and `SPU ID=7`

### CODE REVIEW

**Review by annibalsilva — CHANGES_REQUESTED** [review]

File: src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeGeneticQuality.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  private String geneticTypeCode;
+
+  @Column(name = "GENETIC_WORTH_CODE", nullable = false, length = 3)
+  private String geneticWorthCode;
+
+  @Column(name = "GENETIC_QUALITY_VALUE", nullable = false, precision = 3, scale = 1)
+  private BigDecimal geneticQualityValue;
+
+  @Column(name = "GENETIC_WORTH_CALC_IND", nullable = false)
+  private Character geneticWorthCalcInd;
```
[the remark this answers is not in the document]
Comment (RMCampos) [reply src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeGeneticQuality.java:38]: Nice! I didn't know this one. Thanks!

File: src/main/java/ca/bc/gov/backendstartapi/service/OrchardService.java [hunk 2]
```diff
@@ -44,17 +66,129 @@ public Optional<OrchardLotTypeDescriptionDto> findNotRetiredOrchardValidLotType(
...
+      parentTreeDto.setParentTreeNumber(parentTree.getParentTreeNumber());
+      parentTreeDto.setParentTreeRegStatusCode(parentTree.getParentTreeRegStatusCode());
+      parentTreeDto.setLocalNumber(parentTree.getLocalNumber());
+      parentTreeDto.setActive(parentTree.getActive());
+      parentTreeDto.setTested(parentTree.getTested());
+      parentTreeDto.setBreedingProgram(parentTree.getBreedingProgram());
+      parentTreeDto.setFemaleParentTreeId(parentTree.getFemaleParentParentTreeId());
+      parentTreeDto.setMaleParentTreeId(parentTree.getMaleParentParentTreeId());
+
+      // remove from for
```
Comment (annibalsilva) [inline src/main/java/ca/bc/gov/backendstartapi/service/OrchardService.java:158]: Is this a `TODO`/`FIXME`? If so, we'd be better off marking it as such:

```suggestion
      // TODO: remove from for
```

File: src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeOrchard.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+@Entity
+@Table(name = "PARENT_TREE_ORCHARD")
+public class ParentTreeOrchard {
+
+  @Id
+  @Column(name = "PARENT_TREE_ID")
+  private Long parentTreeId;
+
+  @Column(name = "ORCHARD_ID")
+  private String orchardId;
```
Comment (annibalsilva) [inline src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeOrchard.java:24]: This entity has a composed primary key (checked on the test database), with both `PARENT_TREE_ID` and `ORCHARD_ID` being needed to uniquely identify a registry. Even if `ParentTreeOrchardRepository.findById()` works with this mapping, I think we'd be misleading future developers.

File: src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeGeneticQuality.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  @Id
+  @Column(name = "GENETIC_QUALITY_ID")
+  private Long id;
+
+  @Column(name = "PARENT_TREE_ID", nullable = false)
+  private Long parentTreeId;
+
+  @Column(name = "SEED_PLAN_UNIT_ID", nullable = false)
+  private Long seedPlaningUnitId;
```
[the remark this answers is not in the document]
Comment (RMCampos) [reply src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeGeneticQuality.java:28]: Good catch! Thanks!

File: src/main/java/ca/bc/gov/backendstartapi/service/OrchardService.java (hunk 2 above)
Comment (RMCampos) [reply src/main/java/ca/bc/gov/backendstartapi/service/OrchardService.java:158]: Oops. Forgot to remove. Done!

File: src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeOrchard.java (hunk 3 above)
Comment (RMCampos) [reply src/main/java/ca/bc/gov/backendstartapi/entity/ParentTreeOrchard.java]: Yeah. Good catch! Updates here now. Thanks

**Review by annibalsilva — APPROVED** [review]

**Pull request merged**